### PR TITLE
providing a parsed DataTable in steps' arguments

### PR DESCRIPTION
**Pull request by Sraime**
providing datable by header in steps' arguments

parsed DataTable in steps arguments

## Motivation/Description of the PR

**Description of this PR, which problem it solves**
A 'parse' method in the DataTable object that provide a simplidied version of it. It also provides methods for using it with headers : 

- raw : returns the table as a 2-D array
- rows : returns the table as a 2-D array, without the first row
- hashes : returns an array of objects where each row is converted to an object (column header is the key)

this work is inspired by the same feature from cucumber-js module : https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/data_table_interface.md

**A link to the corresponding issue (if applicable).**
https://github.com/Codeception/CodeceptJS/issues/2081

## Type of change

- [ ] Breaking changes
- [x] New functionality
- [ ] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [x] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/unit/data/dataTableArgument_test.js  1
- lib/data/dataTableArgument.js  1
         

Clones added
============
- test/unit/data/dataTableArgument_test.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4668993&bid=15626735)

File: lib/data/dataTableArgument.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class DataTableArgument {
+  constructor(gherkinDataTable) {
+    this.rawData = gherkinDataTable.rows.map(row => row.cells.map(cell => cell.value));
```
Comment (DavertMik) [inline lib/data/dataTableArgument.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4668993)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/data/dataTableArgument.js  1
- test/unit/data/dataTableArgument_test.js  1
         

Clones added
============
- test/unit/data/dataTableArgument_test.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4668993&bid=15626735)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/data/dataTableArgument.js  1
- test/unit/data/dataTableArgument_test.js  1
         

Clones added
============
- test/unit/data/dataTableArgument_test.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4668993&bid=15626735)

File: lib/data/dataTableArgument.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class DataTableArgument {
+  constructor(gherkinDataTable) {
+    this.rawData = gherkinDataTable.rows.map((row) => {
+      return row.cells.map(cell => cell.value);
```
Comment (DavertMik) [inline lib/data/dataTableArgument.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4668993)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/unit/data/dataTableArgument_test.js  1
- lib/data/dataTableArgument.js  1
         

Clones added
============
- test/unit/data/dataTableArgument_test.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4668993&bid=15626735)

**Comment by Sraime** [comment]
> Looks good, please add the use case examples into `docs/bdd.md`
> This needs to be documented so everyone could benefit from this feature

Done :)

**Comment by DavertMik** [comment]
Thank you!

**Pull request merged**